pkg/controller/controller.go

```diff
@@ -160,16 +207,27 @@ func (c *notificationController) processResource(resource v1.Object, logEntry *l
 			for _, to := range destinations {
 				if changed := notificationsState.SetAlreadyNotified(trigger, cr, to, true); !changed {
 					logEntry.Infof("Notification about condition '%s.%s' already sent to '%v'", trigger, cr.Key, to)
+					eventSequence.addDelivered(NotificationDelivery{
+						Trigger:         trigger,
+						Destination:     to,
+						AlreadyNotified: true,
+					})
 				} else {
 					logEntry.Infof("Sending notification about condition '%s.%s' to '%v'", trigger, cr.Key, to)
 					if err := api.Send(un.Object, cr.Templates, to); err != nil {
 						logEntry.Errorf("Failed to notify recipient %s defined in resource %s/%s: %v",
 							to, resource.GetNamespace(), resource.GetName(), err)
 						notificationsState.SetAlreadyNotified(trigger, cr, to, false)
 						c.metricsRegistry.IncDeliveriesCounter(trigger, to.Service, false)
+						eventSequence.addError(fmt.Errorf("failed to deliver notification %s to %s: %v", trigger, to.Service, err))
```

Comment:
Suggest including the entire `to` object in the log, which includes both `to.Service` and `to.Recipient`.

Reply:
Done